Storage/Site.swift

```diff
@@ -74,7 +74,7 @@ public class Site: Identifiable {
      // Sites may have multiple favicons. We'll return the largest.
     public var icon: Favicon?
     public var latestVisit: Visit?
-    public let bookmarked: Bool?
+    public private(set) var bookmarked: Bool?
 
     public convenience init(url: String, title: String) {
         self.init(url: url, title: title, bookmarked: false)
```

Comment:
`basicHistoryColumnFactory` which is used when we fetch Top Sites from the DB, and also used when we fetch anything from history explicitly uses the designated initialiser and a bookmarked value from the DB. The only usage of this convenience initialiser is not in anything used by the `ActivityStreamPanel`. The `DefaultSuggestedSites` which is the case that we are trying to catch with the nil check passes `nil` into the designated initializer: SuggestedSites.swift line 22
```
super.init(url: data.url, title: data.title, bookmarked: nil)
```